dali/kernels/imgproc/resample/resampling_batch.cu

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cuda_runtime.h>
+#include "dali/kernels/imgproc/resample/resampling_batch.h"
+#include "dali/kernels/imgproc/resample/bilinear_impl.cuh"
+#include "dali/kernels/imgproc/resample/nearest_impl.cuh"
+#include "dali/kernels/imgproc/resample/resampling_impl.cuh"
+
+namespace dali {
+namespace kernels {
+
+template <int which_pass, typename Output, typename Input>
+__global__ void BatchedSeparableResampleKernel(Output *out, const Input *in,
+    const SeparableResamplingSetup::SampleDesc *samples, const SampleBlockInfo *block2sample) {
+  // find which part of which sample this block will process
+  SampleBlockInfo sbi = block2sample[blockIdx.x];
+  SeparableResamplingSetup::SampleDesc sample = samples[sbi.sample];
+  int in_stride, out_stride;
+  Output *sample_out;
+  const Input *sample_in;
+  int blocks;
+
+  DeviceArray<int, 2> in_shape, out_shape;
+
+  in_stride = sample.strides[which_pass];
+  out_stride = sample.strides[which_pass+1];
+  sample_in = in + sample.offsets[which_pass];
+  sample_out = out + sample.offsets[which_pass+1];
+  blocks = sample.block_count.pass[which_pass];
+  in_shape = sample.shapes[which_pass];
+  out_shape = sample.shapes[which_pass+1];
+
+  int block = sbi.block_in_sample;
+
+  // 0 = vertical axis, 1 = horizontal axis
+  int axis = (which_pass == 0) ^ (sample.order == SeparableResamplingSetup::VertHorz);
+
+  ResamplingFilterType ftype = sample.filter_type[axis];
+  ResamplingFilter filter = sample.filter[axis];
+  int support = filter.support();
+
+  float scale = in_shape[axis] * (1.0f / out_shape[axis]);
+
+  int block_size = axis == 1 ? blockDim.x : blockDim.y;
+  int size_in_blocks = (out_shape[axis] + block_size - 1) / block_size;
+
+  int start = min(size_in_blocks * block / blocks * block_size, out_shape[axis]);  // NOLINT
+  int end   = min(size_in_blocks * (block + 1) / blocks * block_size, out_shape[axis]);  // NOLINT
+
+  int x0, x1, y0, y1;
+
+  if (axis == 1) {
+    x0 = start;
+    x1 = end;
+    y0 = 0;
+    y1 = out_shape[0];
+  } else {
+    x0 = 0;
+    x1 = out_shape[1];
+    y0 = start;
+    y1 = end;
+  }
+
+  switch (ftype) {
+  case ResamplingFilterType::Nearest:
+    if (axis == 1) {
+      NNResample(x0, x1, y0, y1, 0, 0, scale, 1, sample_out, out_stride, sample_in, in_stride,
```

Comment:
Do we need to have src_x0 and src_y0 as arguments since they are systematically set to 0?

Reply:
They won't be in fused ResizeCrop.

Reply:
Also, negative scale should do the trick with fused flip.